client/src/js/samples/components/Create/Create.js

```diff
@@ -66,51 +75,62 @@ class CreateSample extends React.Component {
         this.props.onFindFiles();
     };
 
+    handleHide = () => {
+        this.props.onHide();
+        if (this.state.error) {
+            this.props.onClearError("UPDATE_SAMPLE_ERROR");
+        }
+    };
+
     handleModalExited = () => {
-        this.setState({
-            selected: [],
-            name: "",
-            host: "",
-            isolate: "",
-            locale: "",
-            group: this.props.forceGroupChoice ? "none" : "",
-            errors: []
-        });
+        this.setState(getInitialState(this.props));
+    };
+
+    handleChange = (e) => {
+        const name = e.target.name;
+
+        if (name === "name" || name === "subtraction") {
+            const errorType = `error${upperFirst(name)}`;
+
+            this.setState({
+                [name]: e.target.value,
+                [errorType]: ""
+            });
+
+            if (this.state.errorName) {
+                this.props.onClearError("CREATE_SAMPLE_ERROR");
+            }
+        } else {
+            this.setState({
+                [name]: e.target.value
+            });
+        }
+
     };
 
     handleSubmit = (e) => {
```

Comment:
Same thing as earlier. Avoid reusing variables for different values. Instead use `hasError` and directly assign error messages to `state` properties.